# Financial Risk Analysis with Python  
## Task 6: Hypothesis Testing  

**Objective:**  
To statistically test whether high-volume transaction accounts maintain higher average balances compared to low-volume transaction accounts, using hypothesis testing based on customer segmentation.


In [1]:
# Load cleaned data
import pandas as pd

df = pd.read_csv("cleaned_transactions.csv")
df['TransactionDate'] = pd.to_datetime(df['TransactionDate'])

# Calculate total transaction volume per account
txn_volume = (
    df.groupby('AccountID')['TransactionAmount']
      .sum()
      .reset_index(name='total_transaction_volume')
)

# Calculate average account balance per account
avg_balance = (
    df.groupby('AccountID')['AccountBalance']
      .mean()
      .reset_index(name='avg_balance')
)

# Merge volume and balance data
account_data = txn_volume.merge(avg_balance, on='AccountID')

# Define high-volume and low-volume using median split
volume_median = account_data['total_transaction_volume'].median()

account_data['volume_segment'] = account_data['total_transaction_volume'].apply(
    lambda x: 'High Volume' if x >= volume_median else 'Low Volume'
)

account_data.head()


,AccountID,total_transaction_volume,avg_balance,volume_segment
0,ACC10117,599628.73770,94082.590727,High Volume
1,ACC10996,506746.15806,72517.761136,High Volume
2,ACC11062,247384.20924,73541.328302,High Volume
3,ACC11188,127634.92588,51359.039950,Low Volume
4,ACC11285,169318.96953,79923.619870,Low Volume


In [2]:
# Hypothesis testing: High-volume vs Low-volume account balances

from scipy.stats import ttest_ind

# Separate average balances by segment
high_volume_balances = account_data[
    account_data['volume_segment'] == 'High Volume'
]['avg_balance']

low_volume_balances = account_data[
    account_data['volume_segment'] == 'Low Volume'
]['avg_balance']

# Perform independent t-test
t_stat, p_value = ttest_ind(
    high_volume_balances,
    low_volume_balances,
    equal_var=False
)

t_stat, p_value


(1.5629623145199243, 0.12014041906642668)

### Hypothesis Formulation

- **Null Hypothesis (H₀):**  
  There is no significant difference in average account balances between high-volume and low-volume transaction accounts.

- **Alternative Hypothesis (H₁):**  
  High-volume transaction accounts have significantly higher average balances than low-volume transaction accounts.


### Hypothesis Test Result and Interpretation

An independent t-test was conducted to compare the average account balances of high-volume and low-volume transaction accounts. The test resulted in a p-value of 0.1201, which is greater than the significance level of 0.05.

Therefore, the null hypothesis cannot be rejected. This indicates that there is no statistically significant difference in average account balances between high-volume and low-volume transaction accounts.


Although high-volume accounts showed a higher mean balance, the difference was not statistically significant, suggesting that transaction volume alone may not determine average balance levels.
